### PR TITLE
Reorder shouldRedirectToHttps check

### PR DESCRIPTION
**Pull request by jdmcclur**
This check is more efficient in the opposite order.

### CODE REVIEW

**Comment by jdmcclur** [comment]
#build

**Comment by jdmcclur** [comment]
#build

File: dev/com.ibm.ws.webcontainer.security/test/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImplTest.java [hunk 1]
```diff
@@ -793,6 +793,8 @@ private WebRequest createWebRequest(String authMethod, final boolean accessPrecl
                 will(returnValue(accessPrecluded));
                 allowing(webRequest).getApplicationName();
                 will(returnValue(APP_NAME));
+                allowing(webRequest).isSSLRequired();
+                will(returnValue(true));
```
Comment (jhanders34) [inline dev/com.ibm.ws.webcontainer.security/test/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImplTest.java:797]: Does `setHttpServletRequestExpectations` need to be updated as well?  Will `isSecure` be called any longer with your change?

**Review by jhanders34 — APPROVED** [review]
Approving, but you should look at the comment I added and have someone from the servlet / security teams take a look at the change as well.

File: dev/com.ibm.ws.webcontainer.security/test/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImplTest.java (hunk 1 above)
Comment (jdmcclur) [reply dev/com.ibm.ws.webcontainer.security/test/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImplTest.java:797]: I am setting webRequest.isSSLRequired to be true, so isSecured will still be called as before. Note, isSecure is always true for the tests that call shouldRedirectToHttps() which is why webRequest.isSSLRequired was never called before (in the WebAppSecurityCollaboratorImpl tests), as it never made it to the second clause.

**Review by teddyjtorres — APPROVED** [review]

**Pull request merged**